hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/table/action/bootstrap/ParquetBootstrapMetadataHandler.java

```diff
@@ -80,14 +80,15 @@ void executeBootstrap(HoodieBootstrapHandle<?, ?, ?, ?> bootstrapHandle,
         HoodieRecord rec = new HoodieAvroRecord(new HoodieKey(recKey, partitionPath), payload);
         return rec;
       }, table.getPreExecuteRunnable());
-      wrapper.execute();
+      executor.execute();
     } catch (Exception e) {
       throw new HoodieException(e);
     } finally {
-      bootstrapHandle.close();
-      if (null != wrapper) {
-        wrapper.shutdownNow();
+      reader.close();
+      if (null != executor) {
+        executor.shutdownNow();
```

Comment:
any reason we don't awaitTermination here?

Reply:
Finished. And add waiting of termination for other shutdownNow usage.